bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -0,0 +1,348 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.util.VersionInfo;
+
+/** For internal use only - public for technical reasons. */
+@InternalApi("For internal usage only")
+public class BigtableHBaseClassicSettings extends BigtableHBaseSettings {
+
+  public BigtableHBaseClassicSettings(Configuration configuration) {
+    super(configuration);
+  }
+
+  // <editor-fold desc="Public API">
+  /** For internal use only - public for technical reasons. */
+  @InternalApi("For internal usage only")
+  public BigtableOptions getBigtableOptions() throws IOException {
+    BigtableOptions.Builder bigtableOptionsBuilder = BigtableOptions.builder();
+
+    bigtableOptionsBuilder.setProjectId(getValue(PROJECT_ID_KEY, "Project " + "ID"));
+    bigtableOptionsBuilder.setInstanceId(getValue(INSTANCE_ID_KEY, "Instance ID"));
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+
+    if (appProfileId != null) {
+      bigtableOptionsBuilder.setAppProfileId(appProfileId);
+    }
+
+    String dataHostOverride = configuration.get(BIGTABLE_HOST_KEY);
+    if (dataHostOverride != null) {
+      LOG.debug("API Data endpoint host %s.", dataHostOverride);
+      bigtableOptionsBuilder.setDataHost(dataHostOverride);
+    }
+
+    String adminHostOverride = configuration.get(BigtableOptionsFactory.BIGTABLE_ADMIN_HOST_KEY);
+    if (adminHostOverride != null) {
+      LOG.debug("Admin endpoint host %s.", adminHostOverride);
+      bigtableOptionsBuilder.setAdminHost(adminHostOverride);
+    }
+
+    String portOverrideStr = configuration.get(BigtableOptionsFactory.BIGTABLE_PORT_KEY);
+    if (portOverrideStr != null) {
+      bigtableOptionsBuilder.setPort(Integer.parseInt(portOverrideStr));
+    }
+
+    String usePlaintextStr = configuration.get(BIGTABLE_USE_PLAINTEXT_NEGOTIATION);
+    if (usePlaintextStr != null) {
+      bigtableOptionsBuilder.setUsePlaintextNegotiation(Boolean.parseBoolean(usePlaintextStr));
+    }
+
+    setBulkOptions(bigtableOptionsBuilder);
+    setChannelOptions(bigtableOptionsBuilder);
+    setClientCallOptions(bigtableOptionsBuilder);
+
+    String emulatorHost = configuration.get(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY);
+    if (emulatorHost != null) {
+      bigtableOptionsBuilder.enableEmulator(emulatorHost);
+    }
+
+    String useBatchStr = configuration.get(BIGTABLE_USE_BATCH);
+    if (useBatchStr != null) {
+      bigtableOptionsBuilder.setUseBatch(Boolean.parseBoolean(useBatchStr));
+    }
+
+    String useGcjClientStr = configuration.get(BIGTABLE_USE_GCJ_CLIENT);
+    if (useGcjClientStr != null) {
+      bigtableOptionsBuilder.setUseGCJClient(Boolean.parseBoolean(useGcjClientStr));
+    }
+    return bigtableOptionsBuilder.build();
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="Private Helpers">
+  private void setChannelOptions(BigtableOptions.Builder builder) throws IOException {
+    setCredentialOptions(builder);
+
+    builder.setRetryOptions(createRetryOptions());
```

Comment:
retry options are not really channel options. Maybe this should be moved to the caller